egg/zoo/compo_vs_generalization/train.py

```diff
@@ -187,15 +184,22 @@ def main(params):
                            opts.validation_size,
                            rng=rng)
 
-        train, validation, test = [
-            one_hotify(x, opts.n_attributes, opts.n_values)
-            for x in [train, validation, test]
-        ]
+    if opts.masking_mode == 'dedicated_value':
+        # From this line onwards, n_values is added
+        # one value that represents masking the 
+        # corresponding attribute
+        opts.n_values += 1
```

Comment:
I am not sure if this is a safe thing to do. It then becomes a bit harder to tell which mode you're in within the game. At least  I think you should create a new variable like `tot_n_values` rather than overwriting the command line arguments.

Reply:
I felt so bad writing this but it was so convenient :D what you are suggesting seems indeed cleaner